app/reservations/consumer.js

```diff
@@ -0,0 +1,49 @@
+const ampq = require('ampqlib')
+const logger = require('../utils/logger')
+const knex = require('../../db')
+const { EXCHANGE_NAME, NOTIFICATIONS_RK } = require('exchangeConfig')
+const producer = require('../notifications/producer')
+
+class Consumer {
+  static async proceedReservation () {
+    const connection = await ampq.connect('amqp://localhost')
+    const channel = await connection.createChannel()
+    try {
+      await channel.checkExchange(EXCHANGE_NAME)
+    } catch (err) {
+      logger.error(err)
+      process.exit(1)
+    }
+    try {
+      await channel.checkQueue(NOTIFICATIONS_RK)
+    } catch (err) {
+      logger.error(err)
+      process.exit(1)
+    }
+    await channel.consume(NOTIFICATIONS_RK, async (msg) => {
+      let reservationId
+      const startDate = await new Date(msg.reservation.time)
```

Comment:
too much await))
await make sense before function or method that return promise